postreise/plot/plot_powerflow_snapshot.py

```diff
@@ -0,0 +1,342 @@
+        (branch.pf > pf_threshold) & (branch.dist > dist_threshold)
+    ]
+    negative_arrows = branch.loc[
+        (branch.pf < -1 * pf_threshold) & (branch.dist > dist_threshold)
+    ]
+    # Swap direction of negative arrows
+    negative_arrows = negative_arrows.rename(
+        columns={"from_x": "to_x", "to_x": "from_x", "to_y": "from_y", "from_y": "to_y"}
+    )
+    # Finally, plot arrows
+    arrows = pd.concat([positive_arrows, negative_arrows])
+    for i in range(n):
+        start_fraction = i / n
+        end_fraction = (i + 1) / n
+        arrows.apply(
+            lambda a: bokeh_figure.add_layout(
+                Arrow(
+                    end=VeeHead(
+                        line_color="black",
+                        fill_color="gray",
+                        line_width=2,
+                        fill_alpha=0.5,
+                        line_alpha=0.5,
+                        size=a["arrow_size"],
+                    ),
+                    x_start=(a["from_x"] + (a["to_x"] - a["from_x"]) * start_fraction),
+                    y_start=(a["from_y"] + (a["to_y"] - a["from_y"]) * start_fraction),
+                    x_end=(a["from_x"] + (a["to_x"] - a["from_x"]) * end_fraction),
+                    y_end=(a["from_y"] + (a["to_y"] - a["from_y"]) * end_fraction),
+                    line_color=color,
+                    line_alpha=0.7,
+                )
+            ),
+            axis=1,
+        )
+
+
+def aggregate_plant_generation(plant, coordinate_rounding=0):
+    """Aggregate generation for plants based on similar lat/lon coordinates.
+
+    :param int coordinate_rounding: number of digits to round lat & lon for aggregation.
+    :param pandas.DataFrame plant: data frame containing: 'lat', 'lon', 'x', 'y', 'pg'.
+    :return: (*pandas.DataFrame*) -- data frame aggregated to the desired precision.
+    """
+    plant_w_xy = project_bus(plant)
+    plant_w_xy["lat"] = plant_w_xy["lat"].round(coordinate_rounding)
+    plant_w_xy["lon"] = plant_w_xy["lon"].round(coordinate_rounding)
+    aggregated = plant_w_xy.groupby(["lat", "lon"]).agg(
+        {"pg": "sum", "x": "mean", "y": "mean"}
+    )
+    return aggregated
+
+
+def plot_powerflow_snapshot(
+    scenario,
+    hour,
+    b2b_dclines=None,
+    demand_centers=None,
+    ac_branch_color="#8B36FF",
+    dc_branch_color="#01D4ED",
+    solar_color="#FFBB45",
+    wind_color="#78D911",
+    demand_color="gray",
+    figsize=(1400, 800),
+    circle_scale_factor=0.25,
+    bg_width_scale_factor=0.001,
+    pf_width_scale_factor=0.00125,
+    arrow_pf_threshold=3000,
+    arrow_dist_threshold=20,
+    num_ac_arrows=1,
+    num_dc_arrows=1,
+    min_arrow_size=5,
+    branch_alpha=0.5,
+    background_map=False,
+    x_range=None,
+    y_range=None,
+    legend_font_size=None,
+):
+    """Plot a snapshot of powerflow.
+
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario to plot.
+    :param str hour: interval to use for snapshot.
```

Comment:
In other plotting routines, we pass a `pandas.Timestamp/numpy.datetime64/datetime.datetime` as hour to avoid any issue related to weird string formatting when locating the indices. So instead of passing `"2016-11-02 22:00:00",`, the user would pass `pd.Timestamp(2016, 11, 2, 22)`

Reply:
Also, if `hour`is a `pandas.Timestamp/numpy.datetime64/datetime.datetime` then later you can use `_check_date_range_in_scenario` to make sure the query hour is in the scenario